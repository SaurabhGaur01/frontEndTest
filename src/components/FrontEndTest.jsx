import React from 'react';
import Header from './Content/Header';
import Content from './Content/Content';
import makeRequest from '../utils/makeRequest';
import { transformStrings } from '../utils/index';

const FrontEndTest = () => {
    const [listOfCities, updateList] = React.useState({});
    const [selectedCity, setSelectedCity] = React.useState('');
    
    // function to input search box - change event
    const onCityChange = event => {
        const inputCity = event.target.value;
        setSelectedCity(inputCity);
        updateList({});
        if(inputCity.length > 1){
            // make API call to get city list
            makeRequest(event.target.value).then(({ results }) => {
                updateList(results);
            });
        }
    }

    // function to select city from dropdown
    const getSelectedCity = ({ bookingId, name, city, country }) => {
        if(bookingId) {
            setSelectedCity(transformStrings(name, city, country));
            updateList({});
        }
    }

    return (
        <div className="app">
            <Header />
            <Content 
                onCityChange={onCityChange}
                getSelectedCity={getSelectedCity}
                listOfCities={listOfCities}
                selectedCity={selectedCity}
            />
        </div>
    );
}

export default FrontEndTest;
