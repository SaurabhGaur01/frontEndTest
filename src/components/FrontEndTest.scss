.appHeader {
    background: #1879ca;
}

.section {
    position: relative;
}

.backSection {
    position: absolute;
    top: 0;
    z-index: -1;
    width: 100%;
    background: #1879ca;
    height: 20px;
}

.search-bar {
    max-width: 900px;
    margin: 0 auto;
    padding: 0 calc(4px*4);
}

.search-bar__background {
    background: #ffb700;
    border-radius: calc(4px*2);
    box-shadow: 0 calc(4px/2) calc(4px*2) rgba(0,0,0,.16);
}

.searchbox {
    position: relative;
}

.searchbox__suggestions {
    position: absolute;
    top: 42px;
    left: 0;
    animation: fadeIn .5s;
    border: none;
    border-radius: 4px;
    box-shadow: 0 calc(4px/2) calc(4px*2) rgba(0,0,0,.16);
    width: 100%;
}

.searchbox__listing {
    list-style: none;
}

.listingItem {
    cursor: pointer;
}

.iconDisplay {
    height: 25px;
    width: 100px;
}

.textBox {
    font-weight: 700;
}

.listingItem:hover {
    background: lightblue;
}

.form-control:focus{
    border-color: #cccccc;
    -webkit-box-shadow: none;
    box-shadow: none;
}