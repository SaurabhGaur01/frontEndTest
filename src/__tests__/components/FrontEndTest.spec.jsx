import React from 'react';
import { shallow } from 'enzyme';
import FrontEndTest from '../../components/FrontEndTest';

describe('FrontEndTest', () => {
    it('should render correctly', () => {
        const renderedModule = shallow(<FrontEndTest />);
        expect(renderedModule).toMatchSnapshot();
    });
})