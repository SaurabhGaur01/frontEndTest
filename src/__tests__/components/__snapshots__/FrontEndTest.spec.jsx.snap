// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FrontEndTest should render correctly 1`] = `
<div
  className="app"
>
  <Header />
  <Content
    getSelectedCity={[Function]}
    listOfCities={Object {}}
    onCityChange={[Function]}
    selectedCity=""
  />
</div>
`;
